cmd/doctor.go

```diff
@@ -0,0 +1,78 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	vfs "github.com/twpayne/go-vfs"
+)
+
+// doctorCmd represents the doctor command
+var doctorCommand = &cobra.Command{
+	Use:   "doctor",
+	Short: "Check your system for potential problems",
+	Long: `Check your system for potential problems. Such as:
+* Does the configuration file exist
+* Does the destination directory exist
+* Does the source directory exist
+* Is Hashicorp Vault installed?
+* Is Bitwarden installed?
+* Is LastPass installed?
+`,
+	RunE: makeRunE(config.runDoctorCommand),
+}
+
+func init() {
+	rootCommand.AddCommand(doctorCommand)
+
+}
+
+func (c *Config) runDoctorCommand(fs vfs.FS, args []string) error {
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 3, ' ', 0)
+
+	fmt.Fprintf(w, "Arch:\t%s\n", runtime.GOARCH)
+	fmt.Fprintf(w, "OS:\t%s\n", runtime.GOOS)
+
+	for _, dir := range []struct {
+		description string
+		path        string
+	}{
+		{description: "Configuration File", path: configFile},
+		{description: "Destination Directory", path: config.DestDir},
+		{description: "Source Directory", path: config.SourceDir},
+	} {
+		info, err := os.Stat(dir.path)
+		if os.IsNotExist(err) {
+			fmt.Fprintf(w, "%s:\tDoes Not Exist\t(%s)\n", dir.description, dir.path)
+		} else {
+			fmt.Fprintf(w, "%s:\tExists\t(%s %s)\n", dir.description, info.Mode().Perm().String(), dir.path)
+		}
+	}
+
+	for _, binary := range []struct {
+		description string
+		name        string
+	}{
+		{description: "Bitwarden", name: config.Bitwarden.BW},
+		{description: "LastPass", name: config.LastPass.Lpass},
+		{description: "Vault", name: config.Vault.Vault},
+	} {
+		path, err := exec.LookPath(binary.name)
+		if err == nil {
+			info, err := os.Stat(path)
+			if !os.IsNotExist(err) {
+				fmt.Fprintf(w, "%s:\tInstalled\t(%s %s)\n", binary.description, info.Mode().Perm().String(), path)
+			} else {
+				fmt.Fprintf(w, "%s:\tError\t(%s)\n", binary.description, err.Error())
+			}
+		} else {
+			fmt.Fprintf(w, "%s:\tNot Found\n", binary.description)
+		}
```

Comment:
Please use the structure requested for checking directories and files above.

Reply:
I don't believe this is applicable, as described above info is used, and thus is required to be used. If it is, more specificity in the request would be appreciated